docs/Containers.rst

```diff
@@ -57,31 +57,42 @@ that corresponds to the easyconfig files that are specified as arguments to the
 Base container image (``--container-base-image``)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. note:: ``--container-base`` was renamed to `--container-base-image`` in EaysBuild v3.9.2.
+.. note:: ``--container-base`` was renamed to ``--container-base-image`` in EaysBuild v3.9.2.
 
-In order to let EasyBuild generate a container recipe, it is *required* to specify which container image
+In order to let EasyBuild generate a container recipe, you can specify which container image
 should be used as a base, via the ``--container-base-image`` configuration option.
 
-Currently, three types of container base images can be specified:
+Currently, four types of container base images can be specified:
 
-* ``localimage:<path>``: the location of an existing container image file
-* ``docker:<name>``: the name of a Docker container image (to be downloaded from Docker Hub, https://hub.docker.com/)
-* ``shub:<name>``: the name of a Singularity container image (to be downloaded from Singularity Hub, https://singularity-hub.org/)
+* ``library:<spec>``: Singularity container image, hosted on Sylabs Container Library
+* ``localimage:<path>``: path to an existing container image file
+* ``docker:<spec>``: Docker container image, hosted on Docker Hub
+* ``shub:<spec>``: Singularity container image, hosted on Singularity Hub
 
-For the ``docker`` and ``shub`` types, an additional *tag* can be specified: ``docker:<name>:<tag>`` or ``shub:<name>:<tag>``.
+For more details on specifying the container image that should be used, see also :ref:`container_bootstrap_agent_image_based`.
 
 
 .. note:: You can also instruct EasyBuild which container base image should be used via the
-          $EASYBUILD_CONTAINER_BASE_IMAGE environment variable, or by specifying ``container-base-image``
-          in an EasyBuild configuration file;
-          see :ref:`configuration_types`.
+          ``$EASYBUILD_CONTAINER_BASE_IMAGE`` environment variable, or by specifying ``container-base-image``
+          in an EasyBuild configuration file; see :ref:`configuration_types`.
 
-.. note::
-          EasyBuild currently does not (yet) support generating a container recipe that results in a container image
-          that is built from scratch, this will be implemented in a future version of EasyBuild.
-          
-          To get started quickly, we recommend using one of the container base images available from
-          https://singularity-hub.org/collections/143.
+
+.. _containers_usage_container_base_requirements:
+
+Requirements for base container image
++++++++++++++++++++++++++++++++++++++
+
+There are a couple of specific requirements for the base container image:
+
+* all dependencies of EasyBuild must be installed, incl. Lmod (cfr. :ref:`requirements`)
+* a user named ``easybuild`` must be available
+* the ``/scratch`` and ``/app`` directories must exist,
+  and the ``easybuild`` user must have write permissions to those directories
```

Comment:
do we have an example bootstrap file to create such an image to which we can point?

Reply:
I could include a link to https://singularity-hub.org/containers/1569/download/recipe (not sure how long that'll stay around), but the idea is to actually remove these hardcoded requirements in the near term, and i) have support in `eb` to generate a base container from scratch, ii) allow tweaking each of these (username + installation prefix + location for temporary stuff).

Reply:
I would add the link. It's much easier to give a working recipe then to tell what should be done.

Reply:
I'm already using that container in the section on `--container-base-image`, and I'm not a big fan of relying heavily on it.

If things work as planned, we can get rid of these requirements by providing additional keywords in `--container-base-config` for these, so they can be tweaked as needed.

In fact, when you already supply your own template recipe, these are not relevant, since you can control things yourself that way already.